python/ray/serve/_private/metrics_utils.py

```diff
@@ -136,6 +136,12 @@ def add_metrics_point(self, data_points: Dict[Hashable, float], timestamp: float
             # Using in-sort to insert while maintaining sorted ordering.
             bisect.insort(a=self.data[name], x=TimeStampedValue(timestamp, value))
 
+    def prune_data(self, start_timestamp_s: float):
+        """Prune keys that haven't had new data recorded after start_timestamp_s."""
+        for key, datapoints in list(self.data.items()):
+            if len(datapoints) == 0 or datapoints[-1].timestamp < start_timestamp_s:
+                del self.data[key]
```

Comment:
there's probably a better datastructure we can be using for this, e.g., use numpy arrays instead of lists numpy

but for now this is very unlikely to be a bottleneck

Reply:
nit: name it "prune_keys" to make it clear that this _does not_ prune datapoints that are outside of the window if a key is still active